pymc_experimental/model_builder.py

```diff
@@ -0,0 +1,309 @@
+import hashlib
+from pathlib import Path
+from typing import Dict, Union
+
+import arviz as az
+import numpy as np
+import pandas as pd
+import pymc as pm
+
+
+class ModelBuilder(pm.Model):
+    """
+    Extension of pm.Model class to improve workflow.
+    ModelBuilder class can be used to play around models with ease using direct API calls
+    for multiple tasks that one need to deploy a model.
+    """
+
+    _model_type = "BaseClass"
+    version = "None"
+
+    def __init__(
+        self,
+        model_config: Dict,
+        sampler_config: Dict,
+        data: Dict[str, Union[np.ndarray, pd.DataFrame, pd.Series]] = None,
+    ):
+        """
+        Initializes model configuration and sampler configuration for the model
+
+        Parameters
+        ----------
+        model_config : Dictionary
+            dictionary of parameters that initialise model configuration. Generated by the user defined create_sample_input method.
+        sampler_config : Dictionary
+            dictionary of parameters that initialise sampler configuration. Generated by the user defined create_sample_input method.
+        data : Dictionary
+            It is the data we need to train the model on.
+        Examples
+        --------
+        >>> class LinearModel(ModelBuilder):
```

Comment:
there was a totally valid error in the CI
https://github.com/pymc-devs/pymc-experimental/issues/81

The identation in incorrect, the code block should look like this:

```
>>> class LinearModel(ModelBuilder):
>>>     ...
```
not 
```
>>> class LinearModel(ModelBuilder):
>>> ...
```